### PR TITLE
Category media

### PR DESCRIPTION
**Pull request by GeorgosXL**

### CODE REVIEW

File: Parser/MessageParser.php [hunk 1]
```diff
@@ -63,4 +64,46 @@ public function parseMenu(array $data): ?Menu
...
+        try {
+            foreach ($data as $category) {
+                $categories[] = new Category($category);
+                foreach ($this->categoryParsers as $parser) {
+                    $parser->parseCategory($category);
+                }
+            }
+        } catch (\Throwable $t) {
+            $modifiedArray = array_pop($categories);
+            $categories = array_values($modifiedArray); // reindex array
```
Comment (GeorgosXL) [inline Parser/MessageParser.php]: I'd gladly take some feedback here :)

File: Parser/MessageParser.php [hunk 2]
```diff
@@ -63,4 +64,46 @@ public function parseMenu(array $data): ?Menu
...
+            foreach ($data as $media) {
+                $mediaCollection[] = new Media($media);
+                foreach ($this->mediaParsers as $parser) {
+                    $parser->parseMedia($media);
+                }
+            }
+        } catch (\Throwable $t) {
+            $modifiedArray = array_pop($mediaCollection);
+            $mediaCollection = array_values($modifiedArray); // reindex array
+        }
```
Comment (GeorgosXL) [inline Parser/MessageParser.php]: same here

File: Parser/MessageParser.php [hunk 3]
```diff
@@ -68,36 +70,38 @@ public function parseMenu(array $data): ?Menu
...
-        }
-
-        foreach ($this->categoryParsers as $parser) {
-            $parser->parseCategory($category);
+        $categories = [];
+        foreach ($data as $category) {
+            try {
+                $categories[] = new Category($category);
+                foreach ($this->categoryParsers as $parser) {
+                    $parser->parseCategory($category);
```
Comment (Nyholm) [inline Parser/MessageParser.php]: This will not parse the new Category object. Just the array.

File: Parser/MessageParser.php [hunk 4]
```diff
@@ -68,36 +70,38 @@ public function parseMenu(array $data): ?Menu
...
-        try {
-            // TODO decide if it is one or more
-            $category = new Category($data);
-        } catch (\Throwable $t) {
-            return null;
-        }
-
-        foreach ($this->categoryParsers as $parser) {
-            $parser->parseCategory($category);
+        $categories = [];
```
Comment (Nyholm) [inline Parser/MessageParser.php]: You name this `$categories`, but below you are using `$mediaCollection`. Should we be consistent?

File: Parser/MessageParser.php [hunk 5]
```diff
@@ -68,36 +70,38 @@ public function parseMenu(array $data): ?Menu
...
-        }
-
-        foreach ($this->mediaParsers as $parser) {
-            $parser->parseMedia($category);
+        $mediaCollection = [];
+        foreach ($data as $media) {
+            try {
+                $mediaCollection[] = new Media($media);
+                foreach ($this->mediaParsers as $parser) {
+                    $parser->parseMedia($media);
```
Comment (Nyholm) [inline Parser/MessageParser.php]: Same here.

**Pull request merged**